### PR TITLE
feat: Add authz caching config

### PR DESCRIPTION
**Pull request by krithika369**
**What this PR does / why we need it**:

This PR largely replicates the changes for the API in https://github.com/caraml-dev/mlp/pull/87, to have the option of caching authz responses in the XP Management service. As a part of this implementation, the API dependency on MLP has been upgraded.

### CODE REVIEW

File: management-service/services/mlp_service.go [hunk 1]
```diff
@@ -97,7 +97,7 @@ func (service mlpService) refreshProjects() error {
 	ctx, cancel := context.WithTimeout(context.Background(), mlpQueryTimeoutSeconds*time.Second)
 	defer cancel()
 
-	projects, resp, err := service.mlpClient.api.ProjectApi.ProjectsGet(ctx, nil)
+	projects, resp, err := service.mlpClient.api.ProjectApi.V1ProjectsGet(ctx, nil)
```
Comment (krithika369) [inline management-service/services/mlp_service.go:100]: This is a consequence of removing `v1/` from the base path in: https://github.com/caraml-dev/mlp/pull/85

The change was applied to Turing / Merlin already. Eg: https://github.com/caraml-dev/turing/pull/338

**Review by deadlycoconuts — APPROVED** [review]
LGTM! 🚀 Thanks for the changes!

**Comment by deadlycoconuts** [comment]
Just curious, was it intentional to no longer create branches on one's fork for PRs? 😅

**Comment by krithika369** [comment]
> Just curious, was it intentional to no longer create branches on one's fork for PRs? 😅

Since this isn't the practice in Merlin and MLP repos, I decided to just stick to branches for everything (in my new laptop). But I realised that for Turing and XP, the fork's CI steps are cleaner. Will get back to using them for the newer PRs. :)

**Pull request merged**